onnxruntime/core/providers/qnn/builder/onnx_ctx_model_helper.cc

```diff
@@ -89,6 +89,9 @@ Status GetEpContextFromGraph(const onnxruntime::GraphViewer& graph_viewer,
   }
 
   std::string external_qnn_context_binary_file_name = node_helper.Get(EP_CACHE_CONTEXT, "");
+  if (external_qnn_context_binary_file_name.find("..", 0) != std::string::npos) {
```

Comment:
might want to use https://github.com/onnx/onnx/pull/4400 as reference.  their solution is a bit more complete as it normalizes the relative path.  (maybe we don't need to be that complete)
but they also disallow symlink and directories. https://github.com/jnovikov/onnx/blob/40619b54466aafd1bd6a28b25e86fd49f0145be8/onnx/checker.cc#L157